src/js/components/Ballot/BallotScrollingContainer.jsx

```diff
@@ -141,6 +142,26 @@ class BallotScrollingContainer extends Component {
           <CandidateContainer
             data-modal-trigger
             className="u-cursor--pointer"
+            onClick={() => {
+              // DataLayer push
+              const dataLayerObject = {
+                event: 'relatedCandidateCardClick',
+                element_id: oneCandidate.we_vote_id,
+                candidateDetails: {
+                  name: oneCandidate.ballot_item_display_name,
+                  party: oneCandidate.party,
+                  office: oneCandidate.contest_office_name,
+                  state: oneCandidate.state_code,
+                },
+                pageDetails: {
+                  pageName: 'CandidatePage',
+                  section: 'CandidatesWhoRanForSameOffice',
+                  pathname: window.location.pathname,
```

Comment:
To get pathname, instead of `window.location.pathname`, destructuring is preferred:    
const { location: { pathname: currentPathname } } = window;